src/game/TacticalAI/NPC.cc

```diff
@@ -1876,38 +1879,43 @@ void ConverseFull(UINT8 const ubNPC, UINT8 const ubMerc, Approach bApproach, UIN
 				else if ( pQuotePtr->usGiftItem != 0 )
 				{
 					{
-						INT8 bInvPos;
-
 						SOLDIERTYPE* const pSoldier = FindSoldierByProfileID(ubNPC);
+						if (pSoldier)
+						{
+							// Look for item....
+							INT8 const bInvPos = FindObj( pSoldier, pQuotePtr->usGiftItem );
+							if (bInvPos == NO_SLOT) throw std::logic_error("NPC.cc: Gift item does not exist in NPC.");
```

Comment:
Do we really want to throw an exception here? We could just log the error and not gift the item, which would be more robust.

Reply:
It's first and foremost here to tell Coverity that we're really checking our return values, I would not expect to ever see this exception unless somebody breaks something in the future, in which case a game abort might even be preferable to give the problem more urgency.

 ATM it would already assert in a debug build and corrupt memory in release builds if we continue past this point.